lib/resty/wasmx/proxy_wasm.lua

```diff
@@ -38,23 +40,32 @@ local _M = {
     ffi.cdef [[
-        int ngx_http_wasm_ffi_plan_new(ngx_wasm_vm_t *vm,
-                                       ngx_wasm_filter_t *filters,
-                                       size_t n_filters,
-                                       ngx_wasm_plan_t **out,
-                                       u_char *err, size_t *errlen);
-        int ngx_http_wasm_ffi_plan_free(ngx_wasm_plan_t *plan);
-        int ngx_http_wasm_ffi_plan_load(ngx_wasm_plan_t *plan);
-        int ngx_http_wasm_ffi_plan_attach(ngx_http_request_t *r,
-                                          ngx_wasm_plan_t *plan,
-                                          unsigned int isolation);
-        int ngx_http_wasm_ffi_start(ngx_http_request_t *r);
-        int ngx_http_wasm_ffi_set_property(ngx_http_request_t *r,
-                                           ngx_str_t *key,
-                                           ngx_str_t *value);
-        int ngx_http_wasm_ffi_get_property(ngx_http_request_t *r,
-                                           ngx_str_t *key,
-                                           ngx_str_t *out);
+        ngx_int_t ngx_http_wasm_ffi_plan_new(ngx_wasm_vm_t *vm,
+                                             ngx_wasm_filter_t *filters,
+                                             size_t n_filters,
+                                             ngx_wasm_plan_t **out,
+                                             u_char *err, size_t *errlen);
+        ngx_int_t ngx_http_wasm_ffi_plan_free(ngx_wasm_plan_t *plan);
+        ngx_int_t ngx_http_wasm_ffi_plan_load(ngx_wasm_plan_t *plan);
+        ngx_int_t ngx_http_wasm_ffi_plan_attach(ngx_http_request_t *r,
+                                                ngx_wasm_plan_t *plan,
+                                                unsigned int isolation);
+        ngx_int_t ngx_http_wasm_ffi_start(ngx_http_request_t *r);
+        ngx_int_t ngx_http_wasm_ffi_set_property(ngx_http_request_t *r,
+                                                 ngx_str_t *key,
+                                                 ngx_str_t *value);
+        ngx_int_t ngx_http_wasm_ffi_get_property(ngx_http_request_t *r,
+                                                 ngx_str_t *key,
+                                                 ngx_str_t *out);
+        ngx_int_t ngx_http_wasm_ffi_set_host_property(ngx_http_request_t *r,
+                                                      ngx_str_t *key,
+                                                      ngx_str_t *value,
+                                                      unsigned is_const,
+                                                      unsigned retrieve);
+        ngx_int_t ngx_http_wasm_ffi_set_property_setter(ngx_http_request_t *r,
+                                                        ngx_wasm_host_prop_fn_t fn);
+        ngx_int_t ngx_http_wasm_ffi_set_property_getter(ngx_http_request_t *r,
+                                                        ngx_wasm_host_prop_fn_t fn);
```

Comment:
@thibaultcha I still think we should keep the `ngx_int_t` changes because they fix the observable `sizeof` of the return values of these FFI functions to match what's really happening in C... even if most of them are not giving us trouble right now, that might bite us in the future is we keep the discrepancy around.

(As for code aesthetics, I'd be down with moving that string to an unindented top-level thing like
```lua

local ffi_http_cdefs = [[
ngx_int_t ngx_http_wasm_ffi_plan_new(ngx_wasm_vm_t *vm,
...
]]

-- ...


if subsystem == "http" then
    ffi.cdef(ffi_http_cdefs)
```
)

Reply:
So what I encounter because of this `int` to `ngx_int_t` change is that return values (`rc`) from FFI calls such as `C.ngx_http_wasm_ffi_set_host_property` are returned as `cdata` types instead of `number`, which is quite the undesired side-effect imho, especially given `int` should already cover all necessary return codes, just like in the core OpenResty FFI APIs.

Reply:
> So what I encounter because of this int to ngx_int_t change is that return values (rc) from FFI calls such as C.ngx_http_wasm_ffi_set_host_property are returned as cdata types instead of number, which is quite the undesired side-effect imho, especially given int should already cover all necessary return codes, just like in the core OpenResty FFI APIs

Ah, I see. I suspect using `int` here might be working "by accident", just because we're running on little-endian systems.

If we're keeping `int` in the function definitions because it's harmless in practice, I think we need `ngx_int_t` only for the definition of the `ngx_wasm_host_prop_fn_t` function pointer typedef, because that flows the data in the opposite direction (from Lua to C), so LuaJIT compiles that into a C function stub which needs to be 64-bit aware in order to fill the return data with the correct sizeof to be consumed by C code (because otherwise I did see garbage being returned by the callback).